react-components/src/components/CadModelContainer/CadModelContainer.tsx

```diff
@@ -1,24 +1,52 @@
 /*!
  * Copyright 2023 Cognite AS
  */
-import { type ReactElement, useEffect, useRef } from 'react';
-import { type AddModelOptions, type CogniteCadModel } from '@cognite/reveal';
+import { type ReactElement, useEffect, useState } from 'react';
+import {
+  type NodeAppearance,
+  type AddModelOptions,
+  type CogniteCadModel,
+  TreeIndexNodeCollection,
+  NodeIdNodeCollection,
+  DefaultNodeAppearance
+} from '@cognite/reveal';
 import { useReveal } from '../RevealContainer/RevealContext';
 import { type Matrix4 } from 'three';
+import { useSDK } from '../RevealContainer/SDKProvider';
+import { type CogniteClient } from '@cognite/sdk';
 
-type Cognite3dModelProps = {
+export type NodeStylingGroup = {
+  nodeIds: number[];
+  style?: NodeAppearance;
+};
+
+export type TreeIndexStylingGroup = {
+  treeIndices: number[];
+  style?: NodeAppearance;
+};
+
+export type CadModelStyling = {
+  defaultStyle?: NodeAppearance;
+  groups?: Array<NodeStylingGroup | TreeIndexStylingGroup>;
+};
+
+type CogniteCadModelProps = {
   addModelOptions: AddModelOptions;
+  styling?: CadModelStyling;
   transform?: Matrix4;
-  onLoad?: () => void;
+  onLoad?: (model: CogniteCadModel) => void;
 };
 
 export function CadModelContainer({
   addModelOptions,
   transform,
+  styling,
   onLoad
-}: Cognite3dModelProps): ReactElement {
-  const modelRef = useRef<CogniteCadModel>();
+}: CogniteCadModelProps): ReactElement {
+  const [model, setModel] = useState<CogniteCadModel>();
```

Comment:
why `useState` is used instead of `useRef`?

Reply:
Because we need to properly react on it's change and trigger a re-render after that. It's a best practice not to use `someRef.current` as a dependency in `useEffect` hook. See: [link](https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/)